### ISSUE TITLE
Incorrect "Remove unreachable except block" while using Django Rest Framework

### ISSUE THREAD

**Issue by pzelnip** [sourcery-ai/sourcery#266]
### Issue description or question

With this snippet:

```
from rest_framework import serializers
from rest_framework.exceptions import ValidationError

class MySerializer(serializers.Serializer):
    param = serializers.ListField()

s = MySerializer(data={})  # is not valid data
try:
    s.is_valid(raise_exception=True)
except ValueError as exc:
    print('value error')
except ValidationError as exc:
    print('validation error')
```

Sourcery incorrectly marks the 2nd except handler (for `ValidationError`) as unreachable: 

![Screen Shot 2022-08-10 at 11 19 54 AM](https://user-images.githubusercontent.com/414933/183987801-ff6810ee-d848-4662-8024-09f4f2881e36.png)

Yet when I run this program, it prints out "validation error" indicating that handler was reached.  Furthermore after applying the refactor (removing the ValidationError handler) the program now errors with a ValidationError exception:

```
    240 
    241         if self._errors and raise_exception:
--> 242             raise ValidationError(self.errors)
    243 
    244         return not bool(self._errors)

ValidationError: {'param': [ErrorDetail(string='This field is required.', code='required')]}
```

`rest_framework` is the [Django Rest Framework](https://www.django-rest-framework.org/api-guide/exceptions/#validationerror)

### Sourcery Version

VS Code extension v0.12.6

### Code editor or IDE name and version

VS Code:
* Version: 1.70.0
* Commit: da76f93349a72022ca4670c1b84860304616aaa2
* Date: 2022-08-04T04:39:20.716Z (6 days ago)
* Electron: 18.3.5
* Chromium: 100.0.4896.160
* Node.js: 16.13.2
* V8: 10.0.139.17-electron.0
* OS: Darwin arm64 21.5.0

I ran the snippet in a Python 3.9 venv, and the version of DRF installed is 3.11.0.

### OS name and version

OSX Monterey 12.4

**Comment by brendanator** [comment]
Thanks for reporting this @pzelnip, I can reproduce this and it will be fixed in the next release

**Comment by Hellebore** [comment]
Fixed in 0.12.7
